packages/web-components/fast-foundation/src/number-field/number-field.ts

```diff
@@ -169,38 +171,38 @@ export class NumberField extends FormAssociatedNumberField {
     public valueChanged(previousValue, nextValue): void {
         super.valueChanged(previousValue, nextValue);
 
-        const numb = parseFloat(nextValue);
-        let out: number | string = numb == nextValue ? nextValue : numb;
-
-        if (nextValue === "" || isNaN(numb)) {
-            out = "";
-        } else {
-            out = this.getValidValue(numb);
-        }
-
-        this.value = out;
-
-        if (this.proxy instanceof HTMLElement) {
-            this.proxy.value = this.value;
-        }
+        this.updateValue(nextValue);
     }
 
     /**
-     * Ensures that the value is between the min and max values
-     *
-     * @param value - number to evaluate
-     * @returns - a string repesentation
+     * Updates the value. Validates that it's a number, between the min
+     *  and max, updates the proxy and emits events.
      *
+     * @param value - value to be validated
      * @internal
      */
-    private getValidValue(value: number): string {
-        if (this.min !== undefined && value < this.min) {
-            value = this.min;
-        } else if (this.max !== undefined && value > this.max) {
-            value = this.max;
+    private updateValue(value): void {
+        if (value === "" || isNaN(parseFloat(value))) {
+            value = "";
+        } else {
+            value = parseFloat(value);
+            if (this.min !== undefined && value < this.min) {
+                value = this.min;
+            } else if (this.max !== undefined && value > this.max) {
+                value = this.max;
+            }
+
+            value = parseFloat(value.toPrecision(12)).toString();
         }
 
-        return parseFloat(value.toPrecision(12)).toString();
+        if (value != this.value) {
+            this.value = value;
+            if (this.proxy instanceof HTMLElement) {
+                this.proxy.value = this.value;
+            }
+            this.$emit("input");
+            this.$emit("change");
```

Comment:
What does the "input" event mean?  Beforehand it was only emitted when using the step methods, now on every change?

Reply:
HTML number fields fire both the input and change events when ever the user interacts with the field or the step controls. I'm just trying to make both events available per an issue logged. I moved the value updates, validation and event firing to ensure consistency between the different methods to update.